lib/Echidna/SymExec/Verification.hs

```diff
@@ -19,16 +21,32 @@ import Control.Monad.ST (RealWorld)
 
 import Echidna.Types.Campaign (CampaignConf(..), WorkerState)
 import Echidna.Types.Config (Env(..), EConfig(..), OperationMode(..), OutputFormat(..), UIConf(..))
+import Echidna.Types.World (World(..))
 import Echidna.Types.Solidity (SolConf(..))
 import Echidna.Types.Worker (WorkerEvent(..))
-import Echidna.SymExec.Common (rpcFetcher, exploreMethod, TxOrError(..), PartialsLogs)
+import Echidna.SymExec.Common (rpcFetcher, exploreMethod, suitableForSymExec, TxOrError(..), PartialsLogs)
 import Echidna.Worker (pushWorkerEvent)
 
+isSuitableToVerifyMethod :: (MonadIO m, MonadReader Echidna.Types.Config.Env m) => SolcContract -> Method -> [Text.Text] -> m Bool
+isSuitableToVerifyMethod contract method symExecTargets = do
+  env <- ask
+  let allMethods = assocs contract.abiMap
+      assertSigs = env.world.assertSigs
+      (selector, _) = case filter (\(_, m) -> m == method) allMethods of
+        [] -> error $ "Method " ++ show method.name ++ " not found in contract ABI"
+        (x:_) -> x
```

Comment:
something using find is probably more readable, like

```suggestion
      (selector, _) = case find (\(_, m) -> m == method) allMethods of
        Nothing -> error $ "Method " ++ show method.name ++ " not found in contract ABI"
        Just x -> x
```

or some variation with maybe

`maybe (error ".....") snd $ find (snd >>> (== method)) allMethods`

Reply:
Done. I avoided using a one liner, since the error message is already long.